Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCiscoASA.yaml

```diff
@@ -0,0 +1,576 @@
+          , "81","VMTP"
+          , "82","SECURE-VMTP"
+          , "83","VINES"
+          , "84","TTP"
+          , "84","IPTM"
+          , "85","NSFNET-IGP"
+          , "86","DGP"
+          , "87","TCF"
+          , "88","EIGRP"
+          , "89","OSPFIGP"
+          , "90","Sprite-RPC"
+          , "91","LARP"
+          , "92","MTP"
+          , "93","AX.25"
+          , "94","IPIP"
+          , "95","MICP (deprecated)"
+          , "96","SCC-SP"
+          , "97","ETHERIP"
+          , "98","ENCAP"
+          , "99",""
+          , "100","GMTP"
+          , "101","IFMP"
+          , "102","PNNI"
+          , "103","PIM"
+          , "104","ARIS"
+          , "105","SCPS"
+          , "106","QNX"
+          , "107","A/N"
+          , "108","IPComp"
+          , "109","SNP"
+          , "110","Compaq-Peer"
+          , "111","IPX-in-IP"
+          , "112","VRRP"
+          , "113","PGM"
+          , "114",""
+          , "115","L2TP"
+          , "116","DDX"
+          , "117","IATP"
+          , "118","STP"
+          , "119","SRP"
+          , "120","UTI"
+          , "121","SMP"
+          , "122","SM (deprecated)"
+          , "123","PTP"
+          , "124","ISIS over IPv4"
+          , "125","FIRE"
+          , "126","CRTP"
+          , "127","CRUDP"
+          , "128","SSCOPMCE"
+          , "129","IPLT"
+          , "130","SPS"
+          , "131","PIPE"
+          , "132","SCTP"
+          , "133","FC"
+          , "134","RSVP-E2E-IGNORE"
+          , "135","Mobility Header"
+          , "136","UDPLite"
+          , "137","MPLS-in-IP"
+          , "138","manet"
+          , "139","HIP"
+          , "140","Shim6"
+          , "141","WESP"
+          , "142","ROHC"
+          , "143","Ethernet"
+          , "253",""
+          , "254",""
+          , "255","Reserved"
+      ];
+      let NWParser = (
+            starttime:datetime=datetime(null), 
+            endtime:datetime=datetime(null),
+            srcipaddr_has_any_prefix:dynamic=dynamic([]), 
+            dstipaddr_has_any_prefix:dynamic=dynamic([]), 
+            ipaddr_has_any_prefix:dynamic=dynamic([]),
+            dstportnumber:int=int(null), 
+            hostname_has_any:dynamic=dynamic([]), 
+            dvcaction:dynamic=dynamic([]), 
+            eventresult:string='*', 
+            disabled:bool=false)
+        { 
+      let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      let allLogs = CommonSecurityLog
+                | where not(disabled)
+                | where (isnull(starttime) or TimeGenerated >= starttime) and (isnull(endtime) or TimeGenerated <= endtime)
+                | where DeviceVendor == "Cisco" and DeviceProduct == "ASA"
+                | where (isnull(dstportnumber) or (DestinationPort == dstportnumber)) or Message has tostring(dstportnumber)
+                | extend temp_isSrcMatch=has_any_ipv4_prefix(SourceIP,src_or_any), 
```

Comment:
Since this filtering is relevant only to parsed branches, it would fail (i.e. return no results) on unparsed events. The section should be moved to the "parsed" branch.

Reply:
Done